debug: openocd: change how we manage differences between versions

dgloeck opened the pull request:
Every time the array of offsets was changed, it was changed in a backward compatible way by appending elements to the end. But the last time this was done, it was forgotten to raise the version number. On top of that the patch for OpenOCD to support Zephyr has never been updated to support anything else than version 0. So the idea of using a version number to track compatible changes didn't work out.

Therefore add another symbol that can be read by OpenOCD to get the number of elements in the array. This value is automatically calculated during compilation. The version number element should from now on be incremented only for incompatible changes.

Fixes: #13448

dgloeck committed:
debug: openocd: change how we manage differences between versions

Every time the array of offsets was changed, it was changed in a
backward compatible way by appending elements to the end. The version
number was incremented to inform OpenOCD about the existence of the new
elements. But the patch for OpenOCD to support Zephyr has never been
updated to support anything else than version 0, causing OpenOCD to
reject recent Zephyr versions. So the idea of using a version number to
track compatible changes didn't work out.

Therefore add another symbol that can be read by OpenOCD to get the
number of elements in the array. This value is automatically calculated
during compilation. The version number element should from now on be
incremented only for incompatible changes.

Fixes: #13448

Signed-off-by: Daniel Glöckner <dg@emlix.com>

dgloeck commented:
Corresponding changes to zephyrproject-rtos/openocd will follow once I know which branch to base my pull request on.

MaureenHelm reviewed (approved):
> But the last time this was done, it was forgotten to raise the version number.

This isn't quite true. The last change was the second commit in a series, where the first commit in that series incremented the version number. See #10918

dgloeck commented:
@MaureenHelm, I have moved my comment in the array to show that also the floating point offsets can be used if version is 1 and _kernel_openocd_num_offsets does not exist and I have changed the commit message.

MaureenHelm commented:
> @MaureenHelm, I have moved my comment in the array to show that also the floating point offsets can be used if version is 1 and _kernel_openocd_num_offsets does not exist and I have changed the commit message.

Ok, thanks.
